### PR TITLE
Make pylint work with python 3.11 and fix new errors

### PR DESCRIPTION
**Pull request by lmolkova**
Build scripts now use Python 3.11, but old pylint does not work with it:

https://github.com/open-telemetry/build-tools/actions/runs/3439733856/jobs/5737365717

This change updates pylint and also fixes new errors from it.

### CODE REVIEW

File: semantic-conventions/.pylintrc [hunk 1]
```diff
@@ -1,7 +1,7 @@
 [MESSAGES CONTROL]
 # TODO: We should actually fix the warnings instead of disabling them
 #   except for the stuff up to fixme, which is fine to ignore / defer to black. 
-disable = bad-whitespace, wrong-import-order, wrong-hanging-indentation, line-too-long, missing-docstring, unused-import, fixme, invalid-name, too-many-instance-attributes, too-many-locals, too-many-statements, too-many-branches, too-many-arguments, too-many-public-methods, too-few-public-methods, protected-access
```
Comment (Oberon00) [inline semantic-conventions/.pylintrc:4]: Why remove these? That stuff should be handled by black, not pylint.

**Review by Oberon00 — APPROVED** [review]

File: semantic-conventions/.pylintrc (hunk 1 above)
Comment (lmolkova) [reply semantic-conventions/.pylintrc:4]: new pylint complains that it does not know them

**Comment by lmolkova** [comment]
@open-telemetry/specs-approvers can someone please take a look and merge if you're happy with it?

**Comment by Oberon00** [comment]
@open-telemetry/technical-committee I think only you can merge PRs here?

**Comment by arminru** [comment]
@lmolkova Thank you for the fix!
@Oberon00 Thank you for pinging us 🙂

**Pull request merged**